relay-common/src/metrics.rs

```diff
@@ -64,30 +65,96 @@ where
     })
 }
 
+/// Any type that implements this trait can be set as a timer
+/// A typical implementation would be something like this
+///
+/// ```ignore
+///
+///enum MyTimer {
+///    TimeSpentDoingA,
+///    TimeSpentDoingB,
+///}
+///
+///impl TimerMetric for MyTimer {
+///    fn name(&self) -> &str {
+///        match self {
+///            Self::TimeSpentDoingA => "processA.millisecs",
+///            Self::TimeSpentDoingB => "processB.millisecs",
+///        }
+///    }
+///}
+/// ```
+///
+/// The type can then be used with any of the metric timer macros.
+///
+/// ```ignore
```

Comment:
Since this is public, you can merge the examples and make them compile for doc tests:

```suggestion
/// ```
/// use relay_common::{metric, TimerMetric};
```

With this, you no longer need the compile tests at the bottom.

Reply:
Good point